### PR TITLE
refactor: update rounding of minutes

### PR DESCRIPTION
**Pull request by rosvik**
closes https://github.com/AtB-AS/kundevendt/issues/3219

- Adds unit tests for correct rounding of dates
- Implements correct rounding for assistant / departures (see table below)
- Not touching trip search results, since it's being worked on in parallell
- Adds a debug toggle to enable seconds in trip and departure details

<details>
  <summary>Trip and departure details with seconds</summary>
  
simulator_screenshot_4F5E0C4B-AB94-45C4-A1EE-19A7EEF0AF84
  
simulator_screenshot_B7A68095-8060-4D9E-BD75-E124F60204B8

</details>

| Context                             | Journey planner field                             | Expected rounding | Format function                | Touched by this PR |
|-------------------------------------|---------------------------------------------------|----------|--------------------------------|---|
| Front page favorites                | expectedDepartureTime / aimedDepartureTime        | floor    | formatToClockOrRelativeMinutes |   |
| Departures and map                  | expectedDepartureTime / aimedDepartureTime        | floor    | formatToClockOrRelativeMinutes |   |
| Departure details, trip rows        | expectedDepartureTime / aimedDepartureTime        | floor    | formatToClock                  | yes |
| Departure details, header           | aimedDepartureTime (of origin trip row)           | floor    | fullDateTime                   |   |
| Departure details, last passed stop | actualDepartureTime                               | nearest  | formatToClock                  | yes |
| Trip details, leg start             | expectedStartTime / aimedStartTime                | floor    | formatToClock                  | yes |
| Trip details, leg end               | expectedEndTime / aimedEndTime                    | ceil     | formatToClock                  | yes |
| Trip details, last passed stop      | actualDepartureTime                               | nearest  | formatToClock                  | yes |
| Trip search results, legs           | expectedDepartureTime / aimedDepartureTime  | floor    | formatToClock           | yes |
| Trip search results, arrival flag   | expectedDepartureTime / aimedDepartureTime  | ceil     | formatToClock         | yes |

### CODE REVIEW

**Comment by toretefre** [comment]
Looks like https://github.com/AtB-AS/kundevendt/issues/74 might also be closed by this one, good job!

**Comment by rosvik** [comment]
@toretefre This doesn't touch the search results component, as it's being worked on in https://github.com/AtB-AS/kundevendt/issues/3288

File: src/utils/date.ts [hunk 1]
```diff
@@ -418,3 +435,29 @@ function getHumanizer(
...
+
+/**
+ * date-fns also has a rounding function, `roundToNearestMinutes`, but it
+ * doesn't work correctly: https://github.com/date-fns/date-fns/issues/3129
+ *
+ * TODO: Replace with date-fns `roundToNearestMinutes`
+ *
+ * @param date
+ * @param roundingMethod
+ * @returns
```
Comment (gorandalum) [inline src/utils/date.ts:449]:
```suggestion
```

File: src/utils/date.ts [hunk 2]
```diff
@@ -83,9 +83,26 @@ export function secondsBetween(
...
 
-export function formatToClock(isoDate: string | Date, language: Language) {
-  const parsed = isoDate instanceof Date ? isoDate : parseISO(isoDate);
-  return formatLocaleTime(parsed, language);
+/**
+ * @param isoDate
+ * @param language
+ * @param roundingMethod rounding of minutes. Default: ignores seconds
+ * @returns
+ */
```
Comment (gorandalum) [inline src/utils/date.ts:91]: I think it looks a little messy when not all parameters are documented. I think the parameter `roundingMethod` in itself is self-explanatory, but I see the value in describing the default behaviour. Maybe just make roundingMethod an obligatory parameter? I don't think it is too much to ask of callers of this function to be explicit of how they want the minute to be rounded.

**Review by gorandalum — APPROVED** [review]
Very good work, and I really like the way it was presented with the table!

File: src/utils/date.ts (hunk 2 above)
Comment (rosvik) [reply src/utils/date.ts]: Made the param required, and since the the trip search result changes was merged, I added rounding there as well.

**Pull request merged**